### PR TITLE
Flexbox

### LINKED ISSUE #20: Use flexbox to render notifications
- [x] Change how notifications is rendered to flexbox
- [x] Add an oldBrowserSupport-mode to render notifications as table

### PR DESCRIPTION
**Pull request by ja1984**
Fixes #20 

I'm sorry about this pull request as there are a few changes actually.
- Uses flexbox to render each notification, to better handle different size of actionbutton (icon or text)
- Has an fallback where notification is rendered as css table (settings.oldBrowserSupport, defaults to false)
- Have an fix for setting not replacing all settings when using custom setting in init or on notification
- Changed so that dismiss-button can be text or icon, defaults to <i></i> that will render the trash can.

### CODE REVIEW

**Comment by ja1984** [comment]
I have tested it on my computer in Chrome, Edge and Firefox both flexbox and table version seems to be working.

File: index.html [hunk 1]
```diff
@@ -56,27 +60,29 @@ <h2 class="project-tagline">Javascript library to display notifications</h2>
...
-<script src="jackbox.polyfills.js"></script>
-<script src="jackbox.js"></script>
+  <script src="jackbox.polyfills.js"></script>
+  <script src="jackbox.js"></script>
   <script>
 	var notificationCustomSettings = {
 		time : 20,
-		classNames: ["long-notification-class"]
+		classNames: ["long-notification-class"],
+    actionButtonText: 'OK!'
```
Comment (alxlark) [inline index.html]: Please correct indent.

File: jackbox.css [hunk 2]
```diff
@@ -6,17 +6,55 @@
...
+    justify-content: center;
+    transition: height ease .2s;
+    overflow: hidden;
+    height: 0;
+    position: relative;
 }
 
 .notification.show{
-	height: 40px;
+    height: 40px;
```
Comment (alxlark) [inline jackbox.css]: Please use tabs instead of spaces

File: jackbox.css [hunk 3]
```diff
@@ -6,17 +6,55 @@
 }
 
 .notification{
-	padding: 0 10px;
-	height: 0;
-	position: relative;
-	transition: height ease .2s;
-	overflow: hidden;
+    display: flex;
+    width: 100%;
```
Comment (alxlark) [inline jackbox.css]: Please use tabs instead of spaces

File: jackbox.css [hunk 4]
```diff
@@ -6,17 +6,55 @@
...
+    position: relative;
 }
 
 .notification.show{
-	height: 40px;
+    height: 40px;
 }
 
+.notification .icon, .notification .message, .notification .action{
+    flex-grow: 1;
```
Comment (alxlark) [inline jackbox.css]: Please use tabs instead of spaces

File: jackbox.css [hunk 5]
```diff
@@ -6,17 +6,55 @@
...
+
+.notification .icon i{
+    display: block;
+    width: 20px;
+    height: 20px;
+}
+
+.notification .message{
+    font-size: 1.2rem;
+    flex: 3 0px;
```
Comment (alxlark) [inline jackbox.css]: Please delete px near 0

File: jackbox.css [hunk 6]
```diff
@@ -25,35 +63,27 @@
 	background: #fff;
 	opacity: .2;
 	width: 0;
+  flex-grow: none;
```
Comment (alxlark) [inline jackbox.css]: Please use tabs instead of spaces

**Comment by ja1984** [comment]
I don't know what visual studio code have done to my spacing, but I'll fix it!

**Comment by ja1984** [comment]
@alxlark I should have fixed all your comments.

**Removed setting for oldBrowseSupport, it's checked automatically**

**Comment by ja1984** [comment]
Any other input on this or can I merge?

File: index.html [hunk 7]
```diff
@@ -56,27 +57,28 @@ <h2 class="project-tagline">Javascript library to display notifications</h2>
...
-	var notificationCustomSettings = {
-		time : 20,
-		classNames: ["long-notification-class"]
+    var notificationCustomSettings = {
+        time : 20,
+        classNames: ["long-notification-class"],
+        actionButtonText: 'OK!'
 	}
-    var customSettings = {
+    var customSettings = {
```
Comment (alxlark) [inline index.html]: Please clear white space after {

**Comment by alxlark** [comment]
The problem with spaces vs tabs still exist in css file

**Comment by ja1984** [comment]
Must be an issue with my visual studio code then, I removed all spaces and changed to tabs manually.

Do you mind taking a look?

**Comment by alxlark** [comment]
Sure. Also please look at http://stackoverflow.com/questions/29972396/how-to-set-tab-space-style may be it helps

**Comment by ja1984** [comment]
@alxlark thanks man, It looks like mine was set to inserting spaces. Should be fixed now :)

Should I take a stab at this, or will you? :)

**Comment by alxlark** [comment]
If you have time and willing it will be great plus for you, otherwise I will look in the evening

**Comment by ja1984** [comment]
@alxlark looks better now, no?

**Comment by alxlark** [comment]
Yes! 👍

**Pull request merged**